### PR TITLE
262-feat: colorful icons in upcoming courses

### LINKED ISSUE #262: Change b/w icons to colorized
## 📝 Describe the feature

We would like to change b/w course icons to colorized version

## ℹ️ Additional context

Icons examples:

From
Image

To
Image

## ✅ Definition of Done (DoD):

Provide clear criteria that signify the feature's successful implementation.

### PR DESCRIPTION
**Pull request by Wystov**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
In upcoming courses use same colorful icons as in course page
styles converted to module.

## Related Tickets & Documents

- Related Issue #262 
- Closes #262

## Screenshots, Recordings

before
image

after
image

## Added/updated tests?

- [ ] 👌 Yes
- [x] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## Summary by CodeRabbit

- **New Features**
  - Enhanced course display interface with improved styling and image handling.

- **Refactor**
  - Updated `CourseCard` component to use image sources instead of icon elements.
  - Refactored styling approach to use CSS modules and `classNames` for better maintainability.

- **Style**
  - Introduced new stylesheet `courses.module.scss` for consistent and modular styling of course components.

### CODE REVIEW

File: src/widgets/courses-school/ui/courses.scss [hunk 1]
```diff
@@ -58,6 +58,7 @@
           img {
```
Comment (NikitaStarmoussov) [inline src/widgets/courses-school/ui/courses.scss:58]: Please use .module.scss and

Comment (NikitaStarmoussov) [reply src/widgets/courses-school/ui/courses.scss:58]: https://t.me/c/2142251410/1519

File: src/widgets/courses-school/ui/courses.tsx [hunk 2]
```diff
@@ -25,16 +23,14 @@ export const Courses = () => {
     linkLabel = 'More';
   }
 
-  const coursesContent = coursesData?.map(({ title, language, startDate, detailsUrl }) => {
-    const courseIcon = getCourseIcon(title as IconsTitle);
```
[the remark this answers is not in the document]
Comment (Wystov) [reply src/widgets/courses-school/ui/courses.tsx:29]: Yep it was used only here, but I was a little bit scared to remove someone else's code. Will take it out tomorrow

File: src/widgets/courses-school/ui/courses.tsx [hunk 3]
```diff
@@ -1,6 +1,4 @@
 import { CourseCard } from './CourseCard';
-import { getCourseIcon } from '../lib/getCourseIcon';
-import { IconsTitle } from '../lib/icons.data';
```
[the remark this answers is not in the document]
Comment (natanchik) [reply src/widgets/courses-school/ui/courses.tsx:3]: @NikitaStarmoussov Why don't you agree? as far as I can see, we don't need this file anymore and neither do the tsx files with imported icons.

File: src/widgets/courses-school/ui/courses.scss (hunk 1 above)
Comment (Wystov) [reply src/widgets/courses-school/ui/courses.scss:58]: I've just added one line to existed scss file. Or you desided to move whole project on modules and I should change every style file that I'll touch to them?

File: src/widgets/courses-school/ui/courses.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/courses-school/ui/courses.tsx:3]: It's actually used in another place

image

**Review by Yelunina — APPROVED** [review]

**Review by SpaNb4 — APPROVED** [review]

**Pull request merged**